colormaps/core.py

```diff
@@ -161,8 +176,22 @@ def convert(self, data, limits):
         ).filled(self.limits[1] + 1)
         return self.rgba[index]
 
+    def get_legend_data(self, limits, steps):
+        """For a discrete map, we ignore 'steps'."""
+        if limits is None:
+            limits = self.limits
+        else:
+            # tighten custom limits to colormap
+            limits = np.array(limits).clip(*self.limits)
+
+        return np.in1d(
```

Comment:
What do you think of this...?

Reply:
Het is nu geen for loop meer, maar ik snap er ook niets meer van :-)